diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..a49eccd
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(duckdb_parquet_scanner)
+
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+add_executable(a1 main.cpp)
+find_package(duckdb REQUIRED)
+
+target_link_libraries(a1
+    PRIVATE
+        pytick_duckdb_helper
+)
+
diff --git a/cmake/Findduckdb.cmake b/cmake/Findduckdb.cmake
new file mode 100644
index 0000000..0c2843d
--- /dev/null
+++ b/cmake/Findduckdb.cmake
@@ -0,0 +1,56 @@
+include(FetchContent)
+
+# Set policy for DOWNLOAD_EXTRACT_TIMESTAMP
+if(POLICY CMP0135)
+  cmake_policy(SET CMP0135 NEW)
+endif()
+
+# Detect system architecture
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|arm64")
+    set(DUCKDB_ARCH "aarch64")
+else()
+    set(DUCKDB_ARCH "amd64")
+endif()
+
+message(STATUS "Detected architecture: ${DUCKDB_ARCH}")
+
+FetchContent_Declare(
+    duckdb
+    URL ${CMAKE_CURRENT_SOURCE_DIR}/local_data/libduckdb-linux-aarch64.v1.2.2.zip
+    DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+)
+
+FetchContent_MakeAvailable(duckdb)
+find_library(DUCKDB_LIBRARY 
+    NAMES duckdb
+    PATHS ${duckdb_SOURCE_DIR}
+    NO_DEFAULT_PATH
+    REQUIRED
+)
+
+add_library(duckdb::duckdb SHARED IMPORTED)
+set_target_properties(duckdb::duckdb PROPERTIES
+    IMPORTED_LOCATION ${DUCKDB_LIBRARY}
+    INTERFACE_INCLUDE_DIRECTORIES "${duckdb_SOURCE_DIR}"
+)
+
+# FetchContent_Declare(
+#     fmt
+#     URL https://github.com/fmtlib/fmt/archive/refs/tags/11.1.3.zip
+#     DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+# )
+
+# FetchContent_MakeAvailable(fmt)
+
+# Create the DuckDB helper library
+add_library(pytick_duckdb_helper INTERFACE)
+
+set_target_properties(pytick_duckdb_helper PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_SOURCE_DIR}"
+)
+
+target_link_libraries(pytick_duckdb_helper
+    INTERFACE
+        duckdb::duckdb
+        # fmt::fmt
+) 
\ No newline at end of file
diff --git a/main.cpp b/main.cpp
new file mode 100644
index 0000000..48e033b
--- /dev/null
+++ b/main.cpp
@@ -0,0 +1,79 @@
+#include <cstddef>
+#include <cstdint>
+#include <duckdb.h>
+#include <stdio.h>
+#include <string>
+
+int run_query(duckdb_connection conn, const char *query) {
+  duckdb_result result;
+  if (duckdb_query(conn, query, &result) == DuckDBError) {
+    printf("Query 1 error\n");
+    return -1;
+  }
+
+  idx_t duckdb_result_chunk_cnt = duckdb_result_chunk_count(result);
+  printf("result chunk cnt: %ld\n", duckdb_result_chunk_cnt);
+
+  idx_t total_rows = 0;
+  uint64_t largest_nic_time = 0;
+
+  for (idx_t chunk_idx = 0; chunk_idx < duckdb_result_chunk_cnt; chunk_idx++) {
+    duckdb_data_chunk data_chunk = duckdb_result_get_chunk(result, chunk_idx);
+    idx_t data_chunk_size = duckdb_data_chunk_get_size(data_chunk);
+    idx_t column_cnt = duckdb_data_chunk_get_column_count(data_chunk);
+
+    total_rows += data_chunk_size;
+
+    uint64_t *data_vector =
+        (uint64_t *)duckdb_data_chunk_get_vector(data_chunk, 2);
+
+    for (idx_t row_i = 0; row_i < data_chunk_size; row_i += 1) {
+      uint64_t nic_timestamp = data_vector[row_i];
+      if (nic_timestamp > largest_nic_time) {
+        largest_nic_time = nic_timestamp;
+      }
+    }
+
+    // printf("Result chunk idx: %ld, len: %ld, col cnt: %ld\n", chunk_idx,
+    // data_chunk_size, column_cnt);
+  }
+  printf("Total Rows: %ld\n", total_rows);
+  return 0;
+}
+
+int main() {
+
+  // Connect to duckdb
+
+  duckdb_database db;
+  duckdb_connection conn;
+  duckdb_result result;
+
+  duckdb_config config;
+  duckdb_create_config(&config);
+
+  duckdb_set_config(config, "memory_limit", "1GB");
+
+  const char *duckdb_path = "local_data/a.duckdb";
+
+  if (duckdb_open(duckdb_path, &db) == DuckDBError) {
+    printf("Open error: \n");
+    return -1;
+  }
+  if (duckdb_connect(db, &conn) == DuckDBError) {
+    printf("Conenct error\n");
+    return -1;
+  }
+
+  // std::string query = "SELECT * FROM './local_data/*.parquet' LIMIT 10";
+  std::string query = "SELECT * FROM './local_data/*.parquet'";
+  run_query(conn, query.c_str());
+
+  query = "SELECT * FROM './local_data/*.parquet' WHERE ticker != '831'";
+  run_query(conn, query.c_str());
+
+  // Query on all parquet files
+
+  // Test latency on iterations
+  return 0;
+}
\ No newline at end of file
diff --git a/scripts/a.sh b/scripts/a.sh
new file mode 100644
index 0000000..013b386
--- /dev/null
+++ b/scripts/a.sh
@@ -0,0 +1,12 @@
+set -ueo pipefail
+
+mkdir -p build/
+
+cmake -B build/ -H. \
+    -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+    -DCMAKE_BUILD_TYPE=Release
+
+cmake --build build/ 
+
+./build/a1
+
